src/graph/serializers/encoder/with_server_events/v7/encode_graphcontext.c

```diff
@@ -0,0 +1,101 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "encode_v7.h"
+#include "../../../../../util/arr.h"
+
+extern bool process_is_child; // Global variable declared in module.c
+
+// Determine whether we are in the context of a bgsave, in which case
+// the process is independent and should not acquire locks.
+static inline bool _shouldAcquireLocks(void) {
+	return !process_is_child;
+}
+
+static void _RdbSaveHeader(RedisModuleIO *rdb, GraphContext *gc) {
+	/* Header format:
+	* Graph name
+	* Number of graph keys (graph context key + meta keys)
+	* Number of processed keys (current payload index)
+	* Current payload type
+	*/
+
+	// Graph name.
+	RedisModule_SaveStringBuffer(rdb, gc->graph_name, strlen(gc->graph_name) + 1);
+
+	// Number of keys
+	RedisModule_SaveUnsigned(rdb, GraphEncodeContext_GetKeyCount(gc->encoding_context));
+
+	// Payload type
+	RedisModule_SaveUnsigned(rdb, GraphEncodeContext_GetEncodePhase(gc->encoding_context));
+}
+
+// Select the first phase to encode.
+static void _SelectFirstPhase(GraphContext *gc) {
+	// If there are nodes
+	if(Graph_NodeCount(gc->g) > 0) {
+		GraphEncodeContext_SetEncodePhase(gc->encoding_context, NODES);
+	}
+	// If all nodes are deleted
+	else if(array_len(gc->g->nodes->deletedIdx) > 0) {
+		GraphEncodeContext_SetEncodePhase(gc->encoding_context, DELETED_NODES);
+	}
+	// No nodes and no deleted nodes => no edges or deleted edges. Only schema.
+	else {
+		GraphEncodeContext_SetEncodePhase(gc->encoding_context, GRAPH_SCHEMA);
+	}
+}
+
+void RdbSaveGraphContext_v7(RedisModuleIO *rdb, void *value) {
+	/* Encoding format for graph context and graph meta key
+	 * Header
+	 * Payload - Nodes / Edges / Deleted nodes/ Deleted edges/ Graph schema
+	 *
+	 * Payloads encoding order - Nodes => Deleted nodes (if needed) => Edges => Deleted edges (if needed) => Schema.
+	 * Each encoding phase finished
```

Comment:
I find this comment a bit confusing. Feel free to be more verbose than we used to be here; this setup has a lot to pay attention to!